LanguageFeatures/Augmentation-libraries/augmenting_constructors_A03_t01_lib.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - The introductory constructor is `const` and the augmenting constructor is
+///   not or vice versa.
+///
+/// @description Checks that it is a compile-time error if the introductory
+/// constructor is `const` and the augmenting constructor is not.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A03_t01.dart';
+
+augment class C {
+  augment C();
+//        ^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment factory C.foo();
+//                ^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+augment extension type ET {
+  augment ET.foo(this.id);
+//        ^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment factory ET.bar(int id);
```

Comment:
Same problem: Succeeds because we expect and error and get an error, but it is not the expected one.